### ISSUE TITLE
Jupyter Lite breaks after Kernel restart and rerun

### ISSUE THREAD

**Issue by martenrichter** [jupyterlab/jupyterlab#17108]
## Description

If you use Jupyter Lite (Jupyter Lab e.g.) and you push the restart Kernel and rerun button, the kernel status is unknown and it does not work.

## Reproduce

Got to tryJupyter, put a breakpoint to notebooklaspadapter.js at `OnKernelChanged`, you will see it throws at the `untilReady`line. It says cannot read property of undefined (reading widget).....

## Expected behavior

It should not throw.

## Context

I leave this out, as I have already written a patch.

**Comment by jtpio** [comment]
Thanks @martenrichter for the report.

There is also https://github.com/jupyterlite/jupyterlite/pull/1538 which may help with kernel restarts, in case you would like to try the link in the PR, test, or review. Thanks!

**Comment by martenrichter** [comment]
> Thanks @martenrichter for the report.
> 
> There is also [jupyterlite/jupyterlite#1538](https://github.com/jupyterlite/jupyterlite/pull/1538) which should help with kernel restarts, in case you would like to try the link in the PR, test, or review. Thanks!

I may look later. But let'S say, my jupyter related development setup and knowledge is still at its infacy, so I would need to figure out how to integrate it into my development setup.

**Comment by lesteve** [comment]
@martenrichter you can try this [preview link](https://jupyterlite--1538.org.readthedocs.build/en/1538/_static/lab/index.html) of https://github.com/jupyterlite/jupyterlite/pull/1538 and see whether you can reproduce your issue.

**Comment by lesteve** [comment]
> It says cannot read property of undefined (reading widget).....

For completeness, I don't know what that means but as a JupyterLite user I have already seen this warning and I think it can be safely ignored.

**Comment by lesteve** [comment]
To be honest reading the issue description a bit more in details, I am pretty sure this is the same issue as https://github.com/jupyterlite/jupyterlite/issues/1464.

**Comment by jtpio** [comment]
> For completeness, I don't know what that means but as a JupyterLite user I have already seen this warning and I think it can be safely ignored.

Yes it's in the latest versions of lite / lab, but should be fixed by https://github.com/jupyterlab/jupyterlab/pull/17109.

**Comment by martenrichter** [comment]
> To be honest reading the issue description a bit more in details, I am pretty sure this is the same issue as [jupyterlite/jupyterlite#1464](https://github.com/jupyterlite/jupyterlite/issues/1464).

Yes, it should be the same. I am sorry, but I have seen the error made a PR, and then I saw that I should create an issue, so I created one, but did not search for existing...
